sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/UpgradeTest.scala

```diff
@@ -132,20 +125,20 @@ class UpgradeTest extends AnyFreeSpec with Matchers {
     def v1AdditionalDefinitions: String = ""
     def v1ChoiceArgTypeDef: String = "{}"
     def v1AdditionalFields: String = ""
+    def v1AdditionalKeyFields: String = ""
     def v1AdditionalChoices: String = ""
     def v1Precondition: String = "True"
     def v1Signatories: String = s"Cons @Party [Mod:${templateName} {p1} this] (Nil @Party)"
     def v1Observers: String = "Nil @Party"
     def v1Agreement: String = """ "agreement" """
     def v1Key: String =
       s"""
-         |  '$commonDefsPkgId':Mod:Key {
+         |  Mod:${templateName}Key {
          |    label = "test-key",
-         |    maintainers1 = (Cons @Party [Mod:${templateName} {p1} this] (Nil @Party)),
-         |    maintainers2 = (Cons @Party [Mod:${templateName} {p2} this] (Nil @Party))
+         |    maintainers = (Cons @Party [Mod:${templateName} {p1} this] (Nil @Party))
```

Comment:
Repeating this structure could be confusing, It might be worth pulling these first values out into a tuple, and having a `v1AdditionalKeyFieldValues`? I don't know, this is quite the battle of readability

Reply:
I'll give it a try.

Reply:
I did give it a try but it's awkward: we have tests that need to override the entire key expression, like `ChangedKey` or `ThrowingKey`. So I still need `v1Key`, I can't replace it with the proposed `v1AdditionalKeyFieldValues`. Having an overrideable expression (`v1Key`) whose default value in the base class depends on another overridable expression (`v1AdditionalKeyFieldValues`) is pretty error-prone because if a test like `ChangedKey` overrides the former with an expression that does not mention the latter, then overriding the latter has no effect but this isn't obvious at all to the reader of the subclass.